### PR TITLE
Document release process

### PR DESCRIPTION
**Pull request by stsewd**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,10 @@
 release: clean
+	git checkout master
+	git pull origin master
 	python setup.py sdist bdist_wheel
 	python -m twine upload dist/*
+	git tag `python -c "print(__import__('sphinx_search').__version__)"`
+	git push --tags
```
Comment (stsewd) [inline Makefile:7]: Not sure if this is too much automation, let me know what you think

Comment (ericholscher) [reply Makefile:7]: Yea, I don't think we want to autotag & push, I could see issues there with not having incremented the version. Maybe have `tag` be another task?

**Comment by stsewd** [comment]
I have update this PR to make use of another target for tagging, and use something similar to what we have in https://github.com/readthedocs/ethical-ad-server/blob/master/package.json#L9

File: docs/development.rst [hunk 2]
```diff
@@ -22,3 +22,26 @@ Generate minified JS and CSS files via ``Gulp``:
...
+- Run ``npm run build`` and ``npm run changelog``.
+- Open a pull request with the changes.
+
+After the pull request is merged, run ``make release``, this will:
+
+- Checkout and update your master branch.
+- Generate the distribution archives in ``dist/``.
+- Upload the archives from ``dist/`` to PyPI.
+
+Now you can tag the release with ``make tag``.
```
Comment (ericholscher) [inline docs/development.rst:45]: This still seems a bit complicated. We should just have two codeblocks that we can copy & paste exactly that do the release. Basically:

```
::
   npm run build
   npm run changelog

::
   make  release
   make tag

```

Comment (stsewd) [reply docs/development.rst]: :+1: on easy copy/paste, changed

**Review by ericholscher — APPROVED** [review]
I still think we should be standardizing this on our `inv` tasks in `common`. That also auto-increments the versions, which makes it less error prone. I realize thats a big dependency tho, so this approach seems fine for now.

**Pull request merged**